sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -87,20 +87,24 @@ def _create_pipeline_workflow(self, args, pipeline, op_transformers=None, pipeli
     tasks = self._create_dag_templates(pipeline, op_transformers, params)
 
     # generate task reference list for Tekton pipeline
-    task_refs = [
-      {
-        'name': t['metadata']['name'],
-        'taskRef': {
-          'name': t['metadata']['name']
-        },
-        'params': [{
-            'name': p['name'],
-            'value': p.get('default', '')
-          } for p in t['spec'].get('params', [])
-        ]
-      }
-      for t in tasks
-    ]
+    task_refs = []
+    for t in tasks:
+      task_json = {
+          'name': t['metadata']['name'],
+          'taskRef': {
+            'name': t['metadata']['name']
+          },
+          'params': [{
+              'name': p['name'],
+              'value': p.get('default', '')
+            } for p in t['spec'].get('params', [])
+          ]
+        }
+      # add timeout params to task_refs, instead of task.
```

Comment:
@fenglixa -- can you move this code block a little further down? Trying to keep the `task_ref` loop as short as possible. All additional handling of dependencies, parameters, timeout should have their own little code block, ideally in the same order as they appear in the `_op_to_template` method, just to keep some order :-) Thanks